src/server/connection.rs

```diff
@@ -1491,8 +1495,55 @@ impl Connection {
         self.video_ack_required = lr.video_ack_required;
     }
 
+    #[cfg(target_os = "windows")]
+    async fn handle_multiple_user_sessions(&mut self, lr: LoginRequest) -> bool {
+        if let Some(login_request::Union::PortForward(_)) = lr.union {
+            return true;
+        } else {
+            let active_sessions = crate::platform::get_all_active_sessions();
+            if active_sessions.len() <= 1 {
+                return true;
+            }
+            let usid;
+            match lr.option.user_session.parse::<u32>() {
+                Ok(n) => usid = Some(n),
+                Err(..) => usid = None,
+            }
+            let current_process_usid = crate::platform::get_current_process_session_id();
+            if usid.is_none() {
+                let mut res = LoginResponse::new();
+                let mut rdp = Vec::new();
+                for session in active_sessions {
+                    let u_sid = &session[0];
+                    let u_name = &session[1];
+                    let mut rdp_session = RdpUserSession::new();
+                    rdp_session.user_session_id = u_sid.clone();
+                    rdp_session.user_name = u_name.clone();
+                    rdp.push(rdp_session);
+                }
+                res.rdp_user_sessions = rdp;
+                let mut msg_out = Message::new();
+                msg_out.set_login_response(res);
+                self.send(msg_out).await;
+                return false;
```

Comment:
Why is false returned here?

Reply:
I thought the connection stops there so false would be correct. Should I make it true?

Reply:
The server sends all sessions to the client here, and then disconnects itself. I think this may be the reason why the old version reconnects.